### PR TITLE
Fix TODO for calling "on navigate callback"

### PR DESCRIPTION
**Pull request by JensenPaul**
***
<a href="https://pr-preview.s3.amazonaws.com/JensenPaul/fenced-frame/pull/162.html" title="Last updated on May 31, 2024, 6:51 PM UTC (53bac09)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/fenced-frame/162/e67a4f9...JensenPaul:53bac09.html" title="Last updated on May 31, 2024, 6:51 PM UTC (53bac09)">Diff</a>

### CODE REVIEW

File: spec.bs [hunk 1]
```diff
@@ -2837,9 +2840,6 @@ content in the <{fencedframe}> or its embedder, exactly one of two things will h
   :: |automaticBeaconsAllowed|.
 </div>
 
-<span class=XXX>TODO: Call the [=fenced frame config instance/on navigate callback=] at the
```
Comment (gtanzer) [inline spec.bs:2840]: The `on navigate callback` should be called when the urn/config is loaded into a fenced frame, not when there is a top-level navigation.

Comment (JensenPaul) [reply spec.bs:2840]: Fixed.  I misinterpreted Fenced_Frames_Ads_Reporting.md to mean that `reserved.top_navigation_start` is sent on initial ad navigation.

File: spec.bs [hunk 2]
```diff
@@ -2739,6 +2739,8 @@ CORP violation report=] algorithm, as leaving it unfenced may cause a privacy le
 
       1. Set |url| to |config|'s [=fenced frame config/mapped url=]'s [=mapped url/value=].
 
+      1. Run steps in |config|'s [=fenced frame config/on navigate callback=].
```
Comment (domfarolino) [inline spec.bs:2742]:
```suggestion
      1. Run the steps in |config|'s [=fenced frame config/on navigate callback=].
```

**Review by domfarolino — APPROVED** [review]
This looks good to me. The only usage of this appears to be https://wicg.github.io/turtledove/#ref-for-fenced-frame-config-on-navigate-callback%E2%91%A0, and https://wicg.github.io/turtledove/#asynchronously-finish-reporting indeed appears to expect to run in parallel, which is good. Just putting this down in writing for future archaeology if necessary.

**Comment by domfarolino** [comment]
Thank you!

**Pull request merged**